.github/workflows/main.yml

```diff
@@ -0,0 +1,27 @@
+name: Lint and Test
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```

Comment:
Is this filter needed?

Reply:
Not strictly needed, but there are lot of different pull request events that can be triggered: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request. In order to avoid a bunch of extra invocations of the build, it's a good practice to filter explicitly on just the events that we want.

According to the docs, these 3 are the defaults (https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request); so technically it's not needed, but I like things to be explicit.